server/src/test/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequestTests.java

```diff
@@ -134,4 +135,24 @@ public void testSource() throws IOException {
 
         assertEquals(original, processed);
     }
+
+    public void testSkipOperatorOnlyWillNotBeSerialised() throws IOException {
+        RestoreSnapshotRequest original = createTestInstance();
+        assertFalse(original.skipOperatorOnly()); // default is false
+        if (randomBoolean()) {
+            original.skipOperatorOnly(true);
+        }
+        // It is not serialised as xcontent
+        XContentBuilder builder = original.toXContent(XContentFactory.jsonBuilder(), new ToXContent.MapParams(Collections.emptyMap()));
+        XContentParser parser = XContentType.JSON.xContent().createParser(
+            NamedXContentRegistry.EMPTY, null, BytesReference.bytes(builder).streamInput());
+        Map<String, Object> map = parser.mapOrdered();
+        assertFalse(map.containsKey("skip_operator_only"));
```

Comment:
I wonder if it was nicer to verify that `toXContent` gives same result regardless of the `skipOperatorOnly` flag? I.e., invoke it on two instances with different `skipOperatorOnly` flag and verify the maps are identical.

Reply:
Added